### PR TITLE
Support dotnet flavored NuGet projects

### PR DESCRIPTION
**Pull request by osafi**

### CODE REVIEW

**Comment by xtreme-vikram-yadav** [comment]
@osafi Just a reminder that this PR has addressable comments :smiley:

File: lib/license_finder/package_managers/dotnet.rb [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        spec_filename = lib.fetch('files').find { |f| f.end_with?('.nuspec') }
+        return [] if spec_filename.nil?
+
+        @manifest.fetch('packageFolders').keys.map do |root|
+          Pathname(root).join(lib.fetch('path'), spec_filename).to_s
+        end
+      end
+    end
+
+    class PackageMetadata
```
[the remark this answers is not in the document]
Comment (sgravrock) [reply lib/license_finder/package_managers/dotnet.rb]: We added tests for the helper classes.

File: spec/lib/license_finder/package_managers/dotnet_spec.rb [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+          XML
+        end
+
+        before do
+          FileUtils.mkdir_p('packages1')
+          FileUtils.mkdir_p('packages2/thing1 path/5.2.6/')
+          File.write('packages2/thing1 path/5.2.6/thing1 spec.nuspec', thing1_spec)
+        end
+
+        it 'uses the license URL as the license' do
```
[the remark this answers is not in the document]
Comment (sgravrock) [reply spec/lib/license_finder/package_managers/dotnet_spec.rb]: We added tests for the helper.

[the remark this answers is not in the document]
Comment (sgravrock) [reply spec/lib/license_finder/package_managers/dotnet_spec.rb]: We added tests for the helper.

File: spec/lib/license_finder/package_managers/dotnet_spec.rb [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+          licenses = dotnet.current_packages[0].license_names_from_spec
+
+          expect(licenses).to eq(['https://example.com'])
+        end
+      end
+    end
+  end
+
+  describe Dotnet::AssetFile do
+    describe '#packages', skip: LicenseFinder.broken_fakefs? do
```
[the remark this answers is not in the document]
Comment (sgravrock) [reply spec/lib/license_finder/package_managers/dotnet_spec.rb]: Yes. Thanks. Fixed.

File: spec/lib/license_finder/package_managers/dotnet_spec.rb [hunk 4]
```diff
@@ -0,0 +1,198 @@
+require 'spec_helper'
+require 'fakefs/spec_helpers'
+require 'set'
+
+module LicenseFinder
+  def self.broken_fakefs?
```
[the remark this answers is not in the document]
Comment (sgravrock) [reply spec/lib/license_finder/package_managers/dotnet_spec.rb]: Done.

**Comment by xtreme-shane-lattanzio** [comment]
Bumping for webhook trigger.

**Comment by pivotal-issuemaster** [comment]
@osafi Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/pivotal?repositoryId=pivotal/LicenseFinder&pullRequestId=504)!

[Click here](https://cla.pivotal.io/sync/pivotal?repositoryId=pivotal/LicenseFinder&pullRequestId=504) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Pull request merged**